spec/solver_spec.rb

```diff
@@ -0,0 +1,61 @@
+require_relative '../solver'
+
+describe Solver do
+  before :each do
+    @solver = Solver.new
+  end
+
+  it 'shows instance of solver class' do
+    expect(@solver).to be_instance_of Solver
+  end
+
+  context '#factorial' do
+    it 'checks if the factorial method exists' do
+      expect(@solver).to respond_to :factorial
+    end
+
+    it 'gives factorial of given integer' do
+      expect(@solver.factorial(5)).to match 120
+    end
+
+    it 'gives correct value for special case of 0' do
+      expect(@solver.factorial(0)).to match 1
+    end
+
+    it 'gives exception for negative numbers' do
+      expect(@solver.factorial(-5)).to match 'error ! negative number'
```

Comment:
- [ ] [Optional] Thought this works but the ideal way to do it is by using rspec helper method `raise_exception`
```suggestion
      `expect { @resolver.factorial(-5) }.to raise_exception`
```

Reply:
Noted. Thank you very much for your precious review and the time you put to cite the example. It's really appreciated